src/Juvix/Compiler/Core/Extra/Utils.hs

```diff
@@ -38,6 +40,9 @@ substEnvInBranch env br = over caseBranchBody (substEnv env') br
 isClosed :: Node -> Bool
 isClosed = not . has freeVars
 
+mkAxiom :: Interval -> Type -> Node
+mkAxiom loc ty = mkBottom (Info.setInfoLocation loc (Info.setInfoType ty mempty))
```

Comment:
We should probably consider defining functions to shorten this kind of pattern. Perhaps, `setInfoByLocType` or something shorter.

Reply:
I've added a `_bottomType` field to `Bottom`, so now I don't set the type info